generator/src/FluentStageFactoryGenerator.php

```diff
@@ -0,0 +1,163 @@
+        $namespace->addUse(Decimal128::class);
+        $namespace->addUse(Document::class);
+        $namespace->addUse(Int64::class);
+        $namespace->addUse(PackedArray::class);
+        $namespace->addUse(Serializable::class);
+        $namespace->addUse(Timestamp::class);
+        $namespace->addUse(Type::class);
+        $namespace->addUse(ArrayFieldPath::class);
+        $namespace->addUse(FieldPath::class);
+        $namespace->addUse(ResolvesToArray::class);
+        $namespace->addUse(ResolvesToObject::class);
+        $namespace->addUse(Pipeline::class);
+        $namespace->addUse(AccumulatorInterface::class);
+        $namespace->addUse(ExpressionInterface::class);
+        $namespace->addUse(Optional::class);
+        $namespace->addUse(QueryInterface::class);
+        $namespace->addUse(Sort::class);
+        $namespace->addUse(BSONArray::class);
+        $namespace->addUse(stdClass::class);
+        $namespace->addUse(self::FACTORY_CLASS);
+        $class->addProperty('pipeline')
+            ->setType('array')
+            ->setComment('@var list<StageInterface>')
+            ->setValue([]);
+        $class->addMethod('getPipeline')
+            ->setReturnType(Pipeline::class)
+            ->setBody(<<<'PHP'
+                return new Pipeline(...$this->pipeline);
+                PHP);
+
+        $staticFactory = ClassType::from(self::FACTORY_CLASS);
+        assert($staticFactory instanceof ClassType);
+        foreach ($staticFactory->getMethods() as $method) {
+            if (! $method->isPublic()) {
+                continue;
+            }
+
+            try {
+                $this->addMethod($method, $namespace, $class);
+            } catch (Throwable $e) {
+                throw new RuntimeException(sprintf('Failed to generate class for operator "%s"', $operator->name), 0, $e);
+            }
+        }
+
+        $staticFactory = TraitType::from(Stage\FactoryTrait::class);
+        assert($staticFactory instanceof TraitType);
+        foreach ($staticFactory->getMethods() as $method) {
+            if (! $method->isPublic()) {
+                continue;
+            }
+
+            try {
+                $this->addMethod($method, $namespace, $class);
+            } catch (Throwable $e) {
+                throw new RuntimeException(sprintf('Failed to generate class for operator "%s"', $operator->name), 0, $e);
+            }
+        }
+
+        return $namespace;
+    }
+
+    private function addMethod(Method $factoryMethod, PhpNamespace $namespace, ClassType $class): void
+    {
+        if ($class->hasMethod($factoryMethod->getName())) {
+            return;
+        }
+
+        $method = $class->addMethod($factoryMethod->getName());
+
+        $method->setComment($factoryMethod->getComment());
+        $method->setParameters($factoryMethod->getParameters());
+
+        $args = array_map(
+            fn ($param) => '$' . $param->getName(),
+            $factoryMethod->getParameters(),
```

Comment:
Looking at [`FunctionLike::getParameters()`](https://github.com/nette/php-generator/blob/v4.1.4/src/PhpGenerator/Traits/FunctionLike.php#L81), I see this returns an array of `Nette\PhpGenerator\Parameter` objects. Should you use that type-hint in the closure above?

Also, would a `string` return be helpful for static analysis? That's less important to me, as it's clear to the reader that this returns a string via concatenation.

Reply:
Good point. Fixed.